systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py

```diff
@@ -34,16 +34,14 @@ def merge(dbag, data):
     return dbag
 
 
-def search(dbag, name):
+def search(dbag, name, gateway):
     """
     Dirty hack because CS does not deprovision hosts
     """
     hosts = []
     for o in dbag:
         if o == 'id':
             continue
-        print "%s %s" % (dbag[o]['host_name'], name)
-        if dbag[o]['host_name'] == name:
+        print "%s %s %s" % (dbag[o]['host_name'], name, gateway)
+        if dbag[o]['host_name'] == name and dbag[o]['default_gateway'] == gateway :
             hosts.append(o)
-    for o in hosts:
```

Comment:
@rhtyd @ustcweizhou I can't reproduce this on 4.9.2 without the fix. Here's what I've tried:

1. I've created a 2 tier VPC
2. Deployed a VM into Tier1
3. When it started, I've added a new NIC (Tier2) 
4. Restarted the VM
5. VM booted up with no problem, IPs got assigned

![screen shot 2017-05-23 at 3 48 54 pm](https://cloud.githubusercontent.com/assets/13551960/26354927/6d022e56-3fcf-11e7-9c28-69abade2f527.png)